packages/vite/src/node/optimizer/scan.ts

```diff
@@ -310,6 +325,7 @@ async function prepareEsbuildScanner(
     format: 'esm',
     logLevel: 'silent',
     plugins: [...plugins, plugin],
+    jsxDev: !environment.config.isProduction,
```

Comment:
This is to align with esbuild transform

https://github.com/vitejs/vite/blob/f0880df0eab077e61abb533c4edde6538bd729f9/packages/vite/src/node/config.ts#L1508-L1512